### PR TITLE
Move TestAnsibleREX CLI tests to Ansible module

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
### Problem Statement
Ansible-RemoteExecution tests present in REX module under TestAnsibleREX class in CLI 

### Solution
- Move Ansible-RemoteExecution tests present in REX module to Ansible Module in TestAnsibleREX class, to track this tests in correct component module
- Move Ansible-ConfigurationManagement tests present in Ansible to module to new TestAnsibleCfgMgmt class for consistency in CLI test module.
- And few other minor fixes.

### CODE REVIEW

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_ansible.py

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6255
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_ansible.py --external-logging
Test Result : =========== 1 failed, 19 passed, 1519 warnings in 7611.40s (2:06:51) ===========
```

File: tests/foreman/cli/test_ansible.py [hunk 1]
```diff
@@ -4,156 +4,544 @@
...
+        SELECTED_ROLE = 'RedHatInsights.insights-client'
+        SELECTED_ROLE_1 = 'theforeman.foreman_scap_client'
+        SELECTED_VAR = gen_string('alpha')
+        # disable batch tasks to test BZ#2154184
+        target_sat.cli.Settings.set({'name': 'foreman_tasks_proxy_batch_trigger', 'value': 'false'})
+        if rhel_contenthost.os_version.major <= 7:
+            rhel_contenthost.create_custom_repos(rhel7=settings.repos.rhel7_os)
+            assert rhel_contenthost.execute('yum install -y insights-client').status == 0
+        rhel_contenthost.install_katello_ca(target_sat)
+        rhel_contenthost.register_contenthost(module_org.label, force=True)
```
Comment (vijaysawant) [inline tests/foreman/cli/test_ansible.py:84]: `forece` by default set to `True`
```suggestion
        rhel_contenthost.register_contenthost(module_org.label)
```

**Review by vijaysawant — CHANGES_REQUESTED** [review]

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_ansible.py

File: tests/foreman/cli/test_ansible.py (hunk 1 above)
Comment (Gauravtalreja1) [reply tests/foreman/cli/test_ansible.py]: yeah, moved it to Global registration instead

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6260
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_ansible.py --external-logging
Test Result : =========== 1 failed, 19 passed, 1631 warnings in 8138.79s (2:15:38) ===========
```

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_ansible.py -k test_positive_install_ansible_collection

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6267
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_ansible.py::TestAnsibleREX::test_positive_install_ansible_collection --external-logging
Test Result : ================== 1 failed, 4 warnings in 1091.58s (0:18:11) ==================
```

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_ansible.py::TestAnsibleREX::test_positive_install_ansible_collection

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6282
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_ansible.py::TestAnsibleREX::test_positive_install_ansible_collection --external-logging
Test Result : ================== 1 failed, 4 warnings in 1102.53s (0:18:22) ==================
```

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_ansible.py::TestAnsibleREX::test_positive_install_ansible_collection

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6285
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_ansible.py::TestAnsibleREX::test_positive_install_ansible_collection --external-logging
Test Result : ================== 1 passed, 4 warnings in 1052.25s (0:17:32) ==================
```

**Review by shubhamsg199 — APPROVED** [review]
Ack

**Pull request merged**